### ISSUE TITLE
How do you tune the hyper-paramerters?

### ISSUE THREAD

**Issue by JinYang88** [klicperajo/ppnp#8]
I was wondering if my understanding is correct:

You actually create 4 datasets, train, early-stopping, val, test. Then, you tune hyper-parameters on [ train, early-stopping, val], after the best param is obtained by choosing the best-performing one on val set, the param is then applied on the test set (the 4th set) to report the accuracy.

Thanks!

**Comment by gasteigerjo** [comment]
Almost. Additionally, we first split the data into a visible (development) set and a test set. The train, early-stopping and validation sets are drawn from the visible set with varying seeds. Evaluating the model on more than one training data split is essential for GNNs. See Appendix C in the paper for an exact description and illustration.
